gems/smithy-client/spec/smithy-client/endpoint_rules_spec.rb

```diff
@@ -0,0 +1,202 @@
+# frozen_string_literal: true
+
+module Smithy
+  module Client
+    describe EndpointRules do
+      describe '.valid_host_label?' do
+        it 'returns false for an empty value' do
+          expect(EndpointRules.valid_host_label?('')).to be false
+        end
+
+        it 'returns false for values with spaces' do
+          expect(EndpointRules.valid_host_label?('invalid label')).to be false
+        end
+
+        it 'returns true for a valid host label' do
+          expect(EndpointRules.valid_host_label?('valid-label-1')).to be true
+        end
+
+        it 'returns false for values starting with -' do
+          expect(EndpointRules.valid_host_label?('-invalid')).to be false
+        end
+
+        it 'returns false for values ending with -' do
+          expect(EndpointRules.valid_host_label?('invalid-')).to be false
+        end
+
+        it 'returns true for values starting with a number' do
+          expect(EndpointRules.valid_host_label?('1valid')).to be true
+        end
+
+        it 'returns true for a single valid character' do
+          expect(EndpointRules.valid_host_label?('h')).to be true
+        end
+
+        it 'returns false for 64 characters' do
+          expect(EndpointRules.valid_host_label?('h' * 64)).to be false
+        end
+
+        context 'allow_sub_domains=false' do
```

Comment:
Wonky formatting on all of these tests.

Reply:
yea, rubocop -a did something weird to these :-(